### PR TITLE
Coding Challenge Submission

### PR DESCRIPTION
**Pull request by jlocke2**
Hello All!

Here is my submission for the coding challenge.  Checking the first part should be straight forward.  I have included instructions for running the second part in the schedule_builder.rb file at the top.  I look forward to hearing from you and chatting through the various design decisions made.

Thanks again for your time and this opportunity!
Alan

### CODE REVIEW

File: palindrome/tester.rb [hunk 1]
```diff
@@ -1,13 +1,29 @@
...
 
   class T2
     def palindrome?(string)
-      # second implementation
+      # trying something a little different
+      # let's forget strings and just compare arrays
+      letters = string.to_s.downcase.scan(/\w/)
+      return if letters.empty?
+
+      letters == letters.reverse
```
Comment (jason-uh) [inline palindrome/tester.rb]: This isn't technically a different implementation of the algorithm. You're still doing a `reverse` comparison of the characters in the string. The only difference is how you decided to remove non-alphanumeric characters.

Can you come up with a very different algorithm?

Comment (jlocke2) [inline palindrome/tester.rb]: Hey @jason-uh 

Sure.  I could write a recursive or iterative algorithm to compare the end characters working my way to the middle of the string.  I could create a new string object and add letters to it in reverse order of the previous string and compare those two.  I could probably come up with something even more interesting, if you like.  Is this what you had in mind?  Do you want me to add some different options to the pull request?

Comment (jason-uh) [inline palindrome/tester.rb]: Yes, I think either a recursive or iterative algorithm would be good to see. Thanks. Feel free, as well, to change the test cases as you see fit. I purposefully didn't add all edge cases.

Comment (jlocke2) [inline palindrome/tester.rb]: Ok absolutely.  I spent more time focusing on the second part of the coding challenge.  I will spend some more time strengthening this section.  It may not be tonight, but I will definitely have some additions to this up over the weekend.

Comment (jlocke2) [inline palindrome/tester.rb]: Hey @jason-uh.

Just wanted to let you know I had some time this weekend to expand the palindrome section to include a recursive and iterative solution.  I also included some additional tests to catch more edge cases.  Look forward to hearing what you think!

Thanks
Alan
